aten/src/ATen/cuda/CUDAGraph.cpp

```diff
@@ -136,7 +136,12 @@ void CUDAGraph::capture_begin(MempoolId_t pool/*=0*/, cudaStreamCaptureMode capt
   // Addendum: beginAllocateStreamToPool is now called before cudaStreamBeginCapture to prevent an
   // autograd thread's free() call triggering an invalid cudaEventRecord in the caching allocator
   // due to the capture status being updated _after_ a capture had already started.
-  c10::cuda::CUDACachingAllocator::beginAllocateStreamToPool(capture_dev_, capture_stream_, mempool_id_);
+  c10::cuda::CUDACachingAllocator::beginAllocateToPool(capture_dev_, mempool_id_, [this](cudaStream_t stream) {
+      cudaStreamCaptureStatus status;
+      CaptureId_t stream_capture_id;
+      AT_CUDA_CHECK(cudaStreamGetCaptureInfo(stream, &status, &stream_capture_id));
+      return status == cudaStreamCaptureStatus::cudaStreamCaptureStatusActive && stream_capture_id == capture_id_;
```

Comment:
So just to confirm, 

```
with torch.cuda.stream(s0):
  g.capture_begin()
  z = x+y
  with torch.cuda.stream(s1):
    s1.wait_stream(s0)
    w = z+y
  s0.wait_stream(s1)
```

For both the `z` and `w` allocation we will get the same `capture_id_` ?

Reply:
Yes